src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/UpdateFrameMetadata.sql

```diff
@@ -0,0 +1,39 @@
+﻿/*************************************************************
+    Stored procedures for updating an instance status.
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     UpdateFrameMetadata
+--
+-- DESCRIPTION
+--     Update frame metadata for a list of instances matching the watermark.
+--
+-- PARAMETERS
+--     @partitionKey
+--         * The partition key.
+--     @hasFrameMetadata
+--         * flag to indicate frame metadata existance
+--     @watermarkTableType
+--         * The list of watermarks.
+--
+-- RETURN VALUE
+--     None
+--
+CREATE OR ALTER PROCEDURE dbo.UpdateFrameMetadata
+    @partitionKey INT,
+    @hasFrameMetadata BIT,
+	@watermarkTableType dbo.WatermarkTableType READONLY
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SET XACT_ABORT ON
+    BEGIN TRANSACTION
+
+    UPDATE dbo.Instance
+    SET HasFrameMetadata = @hasFrameMetadata
+    FROM dbo.Instance i
+    JOIN @watermarkTableType input ON  i.Watermark = input.Watermark AND i.PartitionKey = @partitionKey
```

Comment:
Is `PartitionKey` only here to use the `IX_Instance_PartitionKey_Watermark` index? If so, why does this index have `PartitionKey`? Isn't `Watermark` unique across all partitions?

Reply:
The index was added by Smitha I believe to keep it consistent across all the sprocs, yes theoretically watermark is unique enough but the index is on partition and watermark